src/viewPanel/sessionWidget.ts

```diff
@@ -27,31 +38,128 @@ export class SessionWidget extends BoxPanel {
 
     this.addWidget(this._tabPanel);
     BoxPanel.setStretch(this._tabPanel, 1);
+
     this._model.tabsChanged.connect(this._onTabsChanged, this);
+
+    this._startKernel();
   }
 
-  private _onTabsChanged(): void {
-    Object.entries(this._model.tabs).forEach(([tabName, tabData], idx) => {
-      const model = new TabModel({
-        tabName,
-        tabData,
-        rendermime: this._rendermime
+  get rendermime(): IRenderMimeRegistry {
+    return this._rendermime;
+  }
+
+  private async _startKernel() {
+    const panel = mockNotebook(this._rendermime, this._context);
+    await this._context?.sessionContext.initialize();
+    await this._context?.sessionContext.ready;
+    // TODO: Make ipywidgets independent from a Notebook context
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this._notebookTracker.widgetAdded.emit(panel);
+    const kernel = this._context?.sessionContext.session?.kernel;
+
+    if (!kernel) {
+      void showDialog({
+        title: 'Error',
+        body: 'Failed to start the kernel for the GLue session',
+        buttons: [Dialog.cancelButton()]
       });
-      const tabWidget = new TabView({ model });
+      return;
+    }
 
-      this._tabPanel.addTab(tabWidget, idx + 1);
-    });
+    this._model.contentsChanged.connect(this._loadData, this);
```

Comment:
We may need to load data once after starting the kernel since the `contentsChanged` signal might be fired before.

```suggestion
    await this._loadData()
    this._model.contentsChanged.connect(this._loadData, this);
```

Reply:
Thanks!